docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,511 @@
+- Add a SEO field (json) to the model. Here, the SEO related tags (title, meta, etc.) will be defined in a _json_ format like so:
+
+```json
+{
+    "title": "MyPage Title",
+    "meta": [
+        {
+            "name": "description",
+            "content":"MyPage Description",
+        },
+        {
+            "property": "og:title",
+            "content": "MyPage Title",
+        },
+        {
+            "property": "og:image",
+            "content": {
+                "id": "6fU8dkL1P9eZlPE9JPw89n"
+            }
+        }
+    ],
+}
+```
+
+- Add a SEO Assets field (many file, media input) to the model.
+- Create a link field type in the models you need SEO and link it to the model you just created.
+
+The resulting content model should look like this.
+
+![SEO Content Model](../../static/img/SEO%20content%20model.png)
+
+In order to obtain the `id` used for meta tags whose content is an asset, you need to select the asset entry and, on the right side panel of the asset, select **Info** and copy the **id**. Later in the application you'll use this **id** to generate the URL for the asset.
+
+On the application itself, the steps to customize the SEO are the following:
+
+- Define a _default_ SEO data.
+- Render the _default_ SEO data in the `App.js` file using [@moxy/next-seo](https://www.npmjs.com/package/@moxy/next-seo) (outside the `Head` tag).
+- Fetch page data as you normally would.
+- Obtain the SEO data from the Contentful page data (example code for the Contentful SEO _parser_)
+
+```js
+import { mapValues } from 'lodash';
+
+// This function will determine if a meta entry's content is an id
+// If it is, it obtains the url to the asset that matches the id
+// Otherwise it returns the content 
+
+const getContent = (content, assets) => {
+    if (isPlainObject(content)) {
+        const asset = assets.find(({ sys }) => sys.id === content.id);
+
+        return asset.fields.file.url.slice(2);
+    }
+
+    return content;
+};
+
+// Provided with the result from the Contentful API call and the entryID for the current page content,
+// This function will extract the SEO data, and complete it with the URL of any needed asset.
+
+const parseContentfulSEO = (contentfulData, entryID) => {
+    const entry = contentfulData.items.find((item) => item.sys.id === entryID);
+    const seoID = entry.fields.seo.id;
+    const seoEntry = entry.includes.Entry.find((entry) => entry.sys.id === seoID);
+    let seoContent;
+
+    try {
+        seoContent = JSON.parse(seoEntry.fields.seo);
+    } catch (parsingError) {
+        console.err('Unvalid JSON in SEO content');
+        console.err(parsingError);
+        return;
+    }
+
+    // Convert every content: { id: ''} into a content: url, keep unchanged otherwise
+    const content = {
+        title: seoContent.title,
+        meta: seoContent.meta.map((entry) => mapValues(entry, (value) => getContent(value, contentfulData.includes.Asset)))
```

Comment:
Missing coma at the end of the line.

I would pass all files through the longer, there are a lot of linting and JS errors in the code blocks.